app/src/Entity/Msgrcpt.php

```diff
@@ -34,4 +40,37 @@ public function setMsgs(?Msgs $msgs): self
 
         return $this;
     }
+
+    public function getAmavisReleaseStartedAt(): ?\DateTimeImmutable
+    {
+        return $this->amavisReleaseStartedAt;
+    }
+
+    public function setAmavisReleaseStartedAt(?\DateTimeImmutable $amavisReleaseStartedAt): static
+    {
+        $this->amavisReleaseStartedAt = $amavisReleaseStartedAt;
+
+        return $this;
+    }
+
+    public function getAmavisReleaseEndedAt(): ?\DateTimeImmutable
+    {
+        return $this->amavisReleaseEndedAt;
+    }
+
+    public function setAmavisReleaseEndedAt(?\DateTimeImmutable $amavisReleaseEndedAt): static
+    {
+        $this->amavisReleaseEndedAt = $amavisReleaseEndedAt;
+
+        return $this;
+    }
+
+    public function isAmavisReleaseStuck(): bool
+    {
+        $tenMinutesAgo = (new \DateTimeImmutable())->modify('-10 minutes');
+
+        return !$this->isAlreadyReleased()
+            && $this->amavisReleaseStartedAt !== null
+            && $this->amavisReleaseStartedAt <= $tenMinutesAgo;
+    }
```

Comment:
My bad sur la proposition, la condition n’était pas bonne (et du coup je suggérais de vérifier si le release était en cours, pas bloqué)

```suggestion
    public function isAmavisReleaseOngoing(): bool
    {
        $tenMinutesAgo = (new \DateTimeImmutable())->modify('-10 minutes');

        return !$this->isAlreadyReleased()
            && $this->amavisReleaseStartedAt !== null
            && $this->amavisReleaseStartedAt >= $tenMinutesAgo;
    }
```